Description: Fix buffer overflow.
 Reserve additional space for final '\0' character.
Author: Leonid Borisenko <leo.borisenko@gmail.com>
Forwarded: http://lists.unbit.it/pipermail/uwsgi/2010-September/000666.html
Applied-Upstream: 0.9.7-dev, http://projects.unbit.it/hg/uwsgi/rev/6c5d6fb3c78d
Last-Update: 2010-09-26

Index: uwsgi-0.9.6/plugins/psgi/psgi_plugin.c
===================================================================
--- uwsgi-0.9.6.orig/plugins/psgi/psgi_plugin.c	2010-09-01 06:48:15.000000000 +0300
+++ uwsgi-0.9.6/plugins/psgi/psgi_plugin.c	2010-09-01 06:48:31.000000000 +0300
@@ -124,7 +124,7 @@
 		goto clear;
         }
 
-	psgibuffer = malloc(stat_psgi.st_size);
+	psgibuffer = malloc(stat_psgi.st_size + 1);
 	if (!psgibuffer) {
 		uwsgi_error("malloc()");
 		close(fd);
